docs: clarify bundling behavior on Schema Object (#653)
asyncapi/spec: spec/asyncapi.md

- format - See [Data Type Formats](#dataTypeFormat) for further details. While relying on JSON Schema's defined formats, the AsyncAPI Specification offers a few additional predefined formats.
- default - The default value represents what would be assumed by the consumer of the input as the value of the schema if one is not provided. Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, of `type` is `string`, then `default` can be `"foo"` but cannot be `1`.

> Alternatively, any time a Schema Object can be used, a [Reference Object](#referenceObject) can be used in its place. This allows referencing definitions in place of defining them inline. It is appropriate to clarify that the `$ref` keyword will follow the behavior described by [Reference Object](#referenceObject) **and not** the one in [JSON Schema definition](https://json-schema.org/understanding-json-schema/structuring.html#ref).

Reviewer: A suggestion to use a more formal language:

```suggestion
Alternatively, any time a Schema Object can be used, a [Reference Object](#referenceObject) can be used in its place. This allows referencing definitions in place of defining them inline. It is appropriate to clarify that the `$ref` keyword MUST follow the behavior described by [Reference Object](#referenceObject) instead of the one in [JSON Schema definition](https://json-schema.org/understanding-json-schema/structuring.html#ref).
```

Author: Good suggestion. Applied!

- default - The default value represents what would be assumed by the consumer of the input as the value of the schema if one is not provided. Unlike JSON Schema, the value MUST conform to the defined type for the Schema Object defined at the same level. For example, of `type` is `string`, then `default` can be `"foo"` but cannot be `1`.

Alternatively, any time a Schema Object can be used, a [Reference Object](#referenceObject) can be used in its place. This allows referencing definitions in place of defining them inline.

> It is appropriate to clarify that the `$ref` keyword will follow the behavior described by [Reference Object](#referenceObject) **and not** the one in JSON Schema definition. Therefore, `$id` keyword won't modify the behavior of `$ref`.

Reviewer 2: To further clarify this problem people will need to wrap their head around what `$id` is and do (we don't describe any of this in the schema object), what JSON Schema is, how it relates to the default super-set of JSON Schema, and this is not relevant to users if they aren't using JSON Schema at all.

Not sure if we can really clarify this, on the other hand, it is kinda needed... Idk 😆 

Did that make sense? 😅

Author: I think I see your point. However, I would like to clarify some things first:

> we don't describe any of this in the schema object

In the "Schema Object" section, we mention it is a superset of JSON Schema Draft-07, including a link to the core and validation documents. Also, we have this statement `As such, any keyword available for those vocabularies is by definition available in AsyncAPI, and will work the exact same way, including but not limited to`

So IMHO it is clear there is a relation and the fact we directly depend on JSON Schema Draft-07 spec.  

Is it enough with just having those lines? Do you suggest we expand the information further?

Reviewer 2: But it also states further down:

> Alternatively, any time a Schema Object can be used, a Reference Object can be used in its place. This allows referencing definitions in place of defining them inline.

Which is even more confusing, when they have nothing to do with each other 😆

Author: I removed the reference to `$id` plus removed the linebreak so it stick all together. Also added a link to the JSON Schema reference docs. Hope that makes more sense now @jonaslagoni !
